### DISCUSSION TITLE
Route Middleware redirects after to hide page is mounted

### DISCUSSION THREAD

**Discussion by sudomk** [nuxt/nuxt#16779] [Questions]
Hi,
using the following middleware the router seems to mount the requested path even if it shouldn't. So it seems like nuxt ist loading and mounting the page and then redirecting to another page.
e.g. I want to visit my login page, but the user is already logged in based on the response of `toSession()`. But login page is still mounted, even though the response is `200`.

Middleware file is: ~/middleware/authentication.global.ts

``` ts
import { useUserStore } from "~~/stores/user"

export default defineNuxtRouteMiddleware(async (from) => {
    const { $ory, $pinia } = useNuxtApp()
    const user = useUserStore($pinia)

    $ory().toSession()
    .then((res) => {
        user.$patch({
            isAuthenticated: true,
            userInfo: res.data.identity.traits
        })

        navigateTo('/success')
    }).catch((err) => {
        user.$reset
        navigateTo('/login')
    })
  })
```

Could anybody tell me what I am doing wrong?

**Comment by sevillaarvin** [comment]
Based on the examples, https://v3.nuxtjs.org/api/utils/define-nuxt-route-middleware/, perhaps you need to return the navigateTo?
